### PR TITLE
Rename bam_sort_samtools to bam_sort_stats_samtools to follow convention

### LINKED ISSUE #2230: new subworkflow bam_sort_samtools rename to bam_sort_stats_samtools

### PR DESCRIPTION
**Pull request by KamilMaliszArdigen**
## PR checklist

Closes #2230  

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: subworkflows/nf-core/align_bowtie2/main.nf [hunk 1]
```diff
@@ -8,7 +8,7 @@ params.samtools_index_options = [:]
 params.samtools_stats_options = [:]
 
 include { BOWTIE2_ALIGN     } from '../../../modules/nf-core/bowtie2/align/main' addParams( options: params.align_options    )
```
Comment (Emiller88) [inline subworkflows/nf-core/align_bowtie2/main.nf:10]:
```suggestion
include { BOWTIE2_ALIGN     } from '../../../modules/nf-core/bowtie2/align/main'
```

File: subworkflows/nf-core/align_bowtie2/main.nf [hunk 2]
```diff
@@ -8,7 +8,7 @@ params.samtools_index_options = [:]
...
-include { BAM_SORT_SAMTOOLS } from '../bam_sort_samtools/main'           addParams( sort_options: params.samtools_sort_options, index_options: params.samtools_index_options, stats_options: params.samtools_stats_options )
+include { BAM_SORT_STATS_SAMTOOLS } from '../bam_sort_stats_samtools/main'           addParams( sort_options: params.samtools_sort_options, index_options: params.samtools_index_options, stats_options: params.samtools_stats_options )
```
Comment (Emiller88) [inline subworkflows/nf-core/align_bowtie2/main.nf:11]:
```suggestion
include { BAM_SORT_STATS_SAMTOOLS } from '../bam_sort_stats_samtools/main'
```
Maybe toss a `nextflow.config` in there

**Review by Emiller88 — CHANGES_REQUESTED** [review]
You want to throw a mermaid diagram in there as well :smiley: ? #2218

File: subworkflows/nf-core/align_bowtie2/main.nf (hunk 1 above)
Comment (KamilMaliszArdigen) [reply subworkflows/nf-core/align_bowtie2/main.nf:10]: Is there any existing example?

**Review by drpatelh — APPROVED** [review]
🎉

**Pull request merged**